tests/fixtures/local_fixtures.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hashlib
+import math
+from dataclasses import dataclass
+from datetime import datetime
+
+import pytest_asyncio  # type: ignore
+import requests  # type: ignore
+from ghga_event_schemas import pydantic_ as event_schemas  # type: ignore
+from ghga_service_chassis_lib.utils import big_temp_file  # type: ignore
+from hexkit.custom_types import JsonObject  # type: ignore
+from hexkit.providers.akafka import KafkaEventPublisher  # type: ignore
+from hexkit.providers.akafka.testutils import EventRecorder  # type: ignore
+from hexkit.providers.s3 import S3Config, S3ObjectStorage  # type: ignore
+
+from tests.fixtures.config import CONFIG
+
+PART_SIZE = 8 * 1024**2
+
+
+@dataclass
+class CheckablePayloadFields:
+    """Fields from recorded event that hold values that can an should be checked"""
+
+    file_id: str
+    part_size: int
+    content_checksum: str
+
+
+@dataclass
+class RecordedEventFixture:
+    """Payload sent and fields that can be checked for equality"""
+
+    payload: JsonObject
+    checkable_fields: CheckablePayloadFields
+
+
+@pytest_asyncio.fixture
+async def local_s3_fixture() -> S3ObjectStorage:
+    """Clean bucket and object"""
+    config = S3Config(
+        s3_endpoint_url=CONFIG.s3_endpoint_url,
+        s3_access_key_id=CONFIG.s3_access_key_id,
+        s3_secret_access_key=CONFIG.s3_access_key_id,
+    )
+    storage = S3ObjectStorage(config=config)
+    if not await storage.does_bucket_exist(bucket_id=CONFIG.inbox_bucket):
+        await storage.create_bucket(bucket_id=CONFIG.inbox_bucket)
+    if await storage.does_object_exist(
+        bucket_id=CONFIG.inbox_bucket, object_id=CONFIG.object_id
+    ):
+        await storage.delete_object(
+            bucket_id=CONFIG.inbox_bucket, object_id=CONFIG.object_id
+        )
+    yield storage
+
+
+@pytest_asyncio.fixture
+async def populated_bucket_fixture(
```

Comment:
Some of the test logic is here in the fixtures, however, the fixtures should only setup and teardown state required for the test to run.